### DISCUSSION TITLE
Mutating immutable obects: with=

### DISCUSSION THREAD

**Discussion by AlwaysHC** [dotnet/csharplang#8465] [Language Ideas]
public record Person(string FistName, string LastName);

Person person = new Person("John", "Doe");

//Changing last name...
person = person with { LastName = "Smith" };
//Proposal...
person with= { LastName = "Smith" };

**Comment by siegfriedpammer** [comment]
The proposed syntax looks awful in my opinion. Also I don't think this feature is worth the effort.

**Comment by 333fred** [comment]
Very similar to https://github.com/dotnet/csharplang/discussions/8114.

**Comment by DavidArno** [comment]
For me, the syntax isn't the issue, it's the idea that making it easy to replace an existing reference to a read-only thing with a reference to a new read-only thing is desirable. This feels like an attempt to work around the fact that record classes are read-only, rather than adding anything useful to the language.

`x = x with { P = newValue }` feels like a code smell to me. Either it's a new thing, so assign it to a new variable, 
`y = x with { P = newValue }` or don't use a record class and  allow the properties to be writable, `x.P = newValue`.
